src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.TargetFrameworkInference.targets

```diff
@@ -96,6 +96,36 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup Condition="'$(TargetFrameworkVersion)' == ''">
     <TargetFrameworkVersion >v0.0</TargetFrameworkVersion>
   </PropertyGroup>
+  
+  <!--
+    Trigger an error if targeting a higher version of .NET Core or .NET Standard than is supported by the current SDK.
+  -->
+
+  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(NETCoreAppMaximumVersion)' == ''">
+    <NETCoreAppMaximumVersion>$(BundledNETCoreAppTargetFrameworkVersion)</NETCoreAppMaximumVersion>
+  </PropertyGroup>
+    
+  <Target Name="_CheckForUnsupportedNETCoreVersion" BeforeTargets="_CheckForInvalidConfigurationAndPlatform;Restore;CollectPackageReferences"
```

Comment:
Running before `Restore` applies to restoring the project from the command line, while running before `CollectPackageReferences` applies to the restore in Visual Studio.  See https://github.com/NuGet/Home/issues/4781.